plugins/inputs/mqtt_consumer/mqtt_consumer.go

```diff
@@ -279,26 +259,72 @@ func (m *MQTTConsumer) recvMessage(_ mqtt.Client, msg mqtt.Message) {
 	}
 }
 
+// compareTopics is used to support the mqtt wild card `+` which allows for one topic of any value
+func compareTopics(expected string, incoming string) bool {
+	expectedSplit := strings.Split(expected, "/")
+	incomingSplit := strings.Split(incoming, "/")
+	if len(expectedSplit) != len(incomingSplit) {
+		return false
+	}
+
+	for i := range expectedSplit {
+		if incomingSplit[i] != expectedSplit[i] && expectedSplit[i] != "+" {
+			return false
+		}
+	}
```

Comment:
Maybe this is nicer to read
```suggestion
	for i, expected := range expectedSplit {
		if incomingSplit[i] != expected && expected != "+" {
			return false
		}
	}
```